xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -345,6 +345,47 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 20]> + Into<[u8; 20]>
 	}
 }
 
+/// Converts a location which is a top-level relay chain (which provides its own consensus) into a 32-byte `AccountId`.
+///
+/// This will always result in the *same account ID* being returned for the same Relay-chain, regardless of the relative security of
+/// this Relay-chain compared to the local chain.
+///
+/// Note: No distinction is made when the local chain happens to be the relay chain in
+/// question or its Relay-chain.
+///
+/// WARNING: This results in the same `AccountId` value being generated regardless
+/// of the relative security of the local chain and the Relay-chain of the input
+/// location. This may not have any immediate security risks, however since it creates
+/// commonalities between chains with different security characteristics, it could
+/// possibly form part of a more sophisticated attack scenario.
```

Comment:
Edited for clarity, however I don't particularly agree with this comment. It does not create commonalities between chains; actually it says that one chain will have the same 32-byte `AccountId` in other consensus systems (that use 32-byte account IDs). If this chain has a security incident, that would have effects in many other chains, but it does not put anything in common between the local chain and the "global" chain.

Unless I am missing something, I would delete this comment.
```suggestion
/// WARNING: This results in the same `AccountId` value being generated regardless of the relative
/// security of the local chain and the security of the input location. This may not have any
/// immediate security risks; however, because it creates commonalities between chains with
/// different security characteristics, it could possibly form part of a more sophisticated attack
/// scenario.
```